src/components/button/button.tsx

```diff
@@ -42,38 +42,51 @@ interface CommonButtonProps {
   className?: string
 }
 
-interface ButtonAsButtonElementProps extends CommonButtonProps, ButtonHTMLAttributes<HTMLButtonElement> {
-  href?: never
-  target?: never
-  disabled?: boolean
-  onClick?: MouseEventHandler<HTMLButtonElement>
+// Define a specialized type for tertiary variant for prop removePadding
+type TertiaryButtonProps = {
+  variant: 'tertiary'
+  removePadding?: boolean // Only tertiary buttons can use removePadding
```

Comment:
The reason for naming this prop `removePadding` is confirmed with Andrei after a discussion with him.